Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -6,7 +6,10 @@ import TuistSupport
 
 public protocol GraphLinting: AnyObject {
     func lint(graphTraverser: GraphTraversing) -> [LintingIssue]
-    func lintCodeCoverageMode(_ mode: CodeCoverageMode?, graphTraverser: GraphTraversing) -> [LintingIssue]
+    func lintCodeCoverageMode(
+        _ mode: Workspace.GenerationOption.AutogeneratedWorkspaceSchemes.CodeCoverageMode,
+        graphTraverser: GraphTraversing
+    ) -> [LintingIssue]
```

Comment:
possibly for another PR to reduce changes here, but seeing the generation option is now part of the `Workspace` we no longer need a dedicated public API for it, the lint check can take place internally within the same `func lint(graphTraverser: GraphTraversing) -> [LintingIssue]`